### DISCUSSION TITLE
Improve WorkBench AutoMapper

### DISCUSSION THREAD

**Discussion by maxpatiiuk** [specify/specify7#5744] [Feature Requests]
- [x] https://github.com/specify/specify7/issues/888
- [ ] Change the architecture to extract indexes out of the header string on the first step rather than do that matching for each header. Examples:
    - `FieldName 1` - > `['FieldName', 1]`
    - `FieldName (12)` - > `['FieldName', 12]`
    - `3 FieldName` - > `['FieldName', 3]`
    - `TableName 1 FieldName` - > `['Table FieldName', 1]` (i.e, `collector 1 first name`)
- [ ] Add a normalization step (i.e, `#`, `no`, `num`, `no.`, `num.`, `number`)
- [ ] Reduce reliance on auto mapper definitions since they break for non-en labels, or if people changed field labels, or if we add new fields for which there are no definitions.

**Comment by bronwyncombs** [comment]
CUIC has reported a need for this. The auto-mapper caused a column for "Collector Last Name" to be uploaded to "Determiner" in a second determination record.
